### PR TITLE
Notifications

### PR DESCRIPTION
**Pull request by michaelFakhry**
Objective
--
[1] Handling Notifications In Strike.
[2] Handling rides view in the base web app.

Details
--
[1] Notifications:
1. check the notifications endpoints in app/adapters/base.js
2. check the header notifications component javascript in app/pods/components/header-notifications/component.js
3. check the header notifications template javascript in app/pods/components/header-notifications/template.hbs
4. check the use of the header-notifications component in app/pods/application/template.hbs

[2] Rides view in the base web app
1. rides the parent route app/pods/base/rides/route.js
2. child route cancelled rides app/pods/base/cancelled-rides/route.js
3. child route claimed rides app/pods/base/claimed-rides/route.js
4. child route confirmed rides app/pods/base/confirmed-rides/route.js
5. child route expired rides app/pods/base/expired-rides/route.js

### CODE REVIEW

File: app/adapters/base.js [hunk 1]
```diff
@@ -18,5 +18,46 @@ export default Ember.Service.extend({
       data: JSON.stringify({userName: username, password: password}),
       url: this.env.APP.apiUrl + this.namespace + '/users/authenticated'
     });
+  },
```
Comment (michaelFakhry) [inline app/adapters/base.js]: notifications endpoints.

File: app/pods/components/header-notifications/component.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
+import Ember from 'ember';
```
Comment (michaelFakhry) [inline app/pods/components/header-notifications/component.js]: the header notifications component

File: app/pods/components/header-notifications/template.hbs [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<div role="presentation" class="dropdown notifications" {{action "viewNotifications"}}>
```
Comment (michaelFakhry) [inline app/pods/components/header-notifications/template.hbs]: header notifications template

File: app/pods/application/template.hbs [hunk 4]
```diff
@@ -49,6 +48,11 @@
               </div>
           {{/link-to}}
 
+
```
Comment (michaelFakhry) [inline app/pods/application/template.hbs]: using the header notification component in the application router

File: app/pods/base/cancelled-rides/route.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
+/**
```
Comment (michaelFakhry) [inline app/pods/base/cancelled-rides/route.js]: handling cancelled-rides

File: app/pods/base/claimed-rides/route.js [hunk 6]
```diff
@@ -0,0 +1,302 @@
+/**
```
Comment (michaelFakhry) [inline app/pods/base/claimed-rides/route.js]: handling claimed rides

File: app/pods/base/confirmed-rides/route.js [hunk 7]
```diff
@@ -0,0 +1,119 @@
+/**
```
Comment (michaelFakhry) [inline app/pods/base/confirmed-rides/route.js]: handling confirmed rides

File: app/pods/base/expired-rides/route.js [hunk 8]
```diff
@@ -0,0 +1,119 @@
+/**
```
Comment (michaelFakhry) [inline app/pods/base/expired-rides/route.js]: expired rides

File: app/pods/base/rides/route.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
+/**
```
Comment (michaelFakhry) [inline app/pods/base/rides/route.js]: the parent route

File: app/pods/components/header-notifications/component.js [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+    })[0];
+
+    //cloning the notification object to avoid overriding it
+    notificationToDisplay = jQuery.extend(true, {}, notificationToDisplay);
+
+    // add the identifier to the message if we have it
+    if (notification.identifier) {
+      notificationToDisplay.message = notificationToDisplay.message.replace('#(identifier)', notification.identifier);
+    }
+
```
Comment (engpeter2010) [inline app/pods/components/header-notifications/component.js]: Remove the duplicated two lines below

File: app/pods/components/ride-form/component.js [hunk 11]
```diff
@@ -795,101 +814,75 @@ export default Ember.Component.extend(EmberValidations, {
...
-    /**
-     * rebidding a ride
-     */
     rebid: function () {
       var self = this;
       var fullRide = this.getRideAfterValidation();
       this.get('getLatLongFromAddress')(fullRide.fullPickUpAddressName).then(function (response) {
         console.log(response);
 
-        if(response.results && response.results.length >0){
```
Comment (engpeter2010) [inline app/pods/components/ride-form/component.js]: Replace all these kind of checks with Ember.isEmpty().

File: app/pods/components/ride-form/component.js [hunk 12]
```diff
@@ -795,101 +814,75 @@ export default Ember.Component.extend(EmberValidations, {
...
-     * submitting a ride
-     */
     submit: function () {
       var self = this;
       var fullRide = this.getRideAfterValidation();
       this.get('getLatLongFromAddress')(fullRide.fullPickUpAddressName).then(function (response) {
-        debugger;
         console.log(response);
-        if(response.results && response.results.length >0){
+        if (response.results && response.results.length > 0) {
```
Comment (engpeter2010) [inline app/pods/components/ride-form/component.js]: Replace this check with Ember.isEmpty(). Do the same for all such checks in the code. Also remove log messages from code.

File: app/pods/components/airport-form/template.hbs [hunk 13]
```diff
@@ -43,7 +43,7 @@
       {{/if}}
     </label>
     <div class="col-md-6 col-sm-6 col-xs-12">
-      {{bs-datetimepicker date=attributes.myTime format='YYYY-MM-DD hh:mm' updateDate=(action (mut attributes.time))}}
+      {{bs-datetimepicker date=attributes.myTime format='DD-MM-YYYY hh:mm' updateDate=(action (mut attributes.time))}}
```
Comment (engpeter2010) [inline app/pods/components/airport-form/template.hbs]: It is better o move this format to environment variables for all datetime pickers

**Pull request merged**